app/publish/publish.go

```diff
@@ -127,7 +150,18 @@ func (h UploadHandler) CreateFile(accountID string, origFilename string) (*os.Fi
 }
 
 func (h UploadHandler) preparePath(accountID string) (string, error) {
-	path := path.Join(h.uploadPath, accountID)
+	path := path.Join(h.UploadPath, accountID)
 	err := os.MkdirAll(path, os.ModePerm)
 	return path, err
 }
+
+// CanHandle checks if http.Request contains POSTed data in an accepted format.
+// Supposed to be used in gorilla mux router MatcherFunc.
+func (h UploadHandler) CanHandle(r *http.Request, _ *mux.RouteMatch) bool {
+	_, _, err := r.FormFile(FileFieldName)
```

Comment:
What happens if it's a different error? What then? We assume we can handle it?

Reply:
If the field is present, we assume we _should_ handle it. It is the job of the main handler routine to deal with whatever other errors might occur.